contract/src/main/java/io/neow3j/contract/types/NNSName.java

```diff
@@ -0,0 +1,144 @@
+package io.neow3j.contract.types;
+
+import io.neow3j.contract.exceptions.InvalidNeoNameException;
+import io.neow3j.contract.exceptions.InvalidNeoNameServiceRootException;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Represents a NeoNameService domain name.
+ */
+public class NNSName {
+
+    private final String name;
+
+    /**
+     * Creates a NNS name and checks its validity.
+     *
+     * @param name the domain name.
+     * @throws InvalidNeoNameException if the format of the provided name is invalid.
+     */
+    public NNSName(String name) throws InvalidNeoNameException {
```

Comment:
> I'm just thinking how I hate having to catch exception in Java, even if I'm sure they will not happen because of other checks.

True, I'll change it to an unchecked exception.